hax-lib/src/int/mod.rs

```diff
@@ -0,0 +1,133 @@
+use core::ops::*;
+use num_traits::cast::ToPrimitive;
+
+mod bigint;
+use bigint::*;
+
+/// Mathematical integers for writting specifications. Mathematical
+/// integers are unbounded and arithmetic operation on them never over
+/// or underflow.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Int(BigInt);
+pub use hax_lib_macros::int;
+
+#[crate::exclude]
+impl Int {
+    fn new(x: impl Into<num_bigint::BigInt>) -> Self {
+        Int(BigInt::new(&x.into()))
+    }
+    fn get(self) -> num_bigint::BigInt {
+        self.0.get()
+    }
+}
+
+impl Add for Int {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self::Output {
+        Self::new(self.get() + other.get())
+    }
+}
+
+impl Sub for Int {
+    type Output = Self;
+
+    fn sub(self, other: Self) -> Self::Output {
+        Self::new(self.get() - other.get())
+    }
+}
+
+impl Mul for Int {
+    type Output = Self;
+
+    fn mul(self, other: Self) -> Self::Output {
+        Self::new(self.get() * other.get())
+    }
+}
+
+impl Div for Int {
+    type Output = Self;
+
+    fn div(self, other: Self) -> Self::Output {
+        Self::new(self.get() / other.get())
+    }
+}
+
+impl Int {
+    /// Raises `2` at the power `self`
+    pub fn pow2(self) -> Self {
```

Comment:
Do we need this function anywhere?

Reply:
Yes, we use that a lot for Kyber and its spec